packages/svelte/src/preprocessors.ts

```diff
@@ -0,0 +1,90 @@
+import MagicString from 'magic-string';
+
+import { ComponentTrackingInitOptions, PreprocessorGroup, TrackComponentOptions } from './types';
+
+export const defaultComponentTrackingOptions: Required<ComponentTrackingInitOptions> = {
+  trackComponents: true,
+  trackInit: true,
+  trackUpdates: true,
+};
+
+/**
+ * Svelte Preprocessor to inject Sentry performance monitoring related code
+ * into Svelte components.
+ */
+export function componentTrackingPreprocessor(options?: ComponentTrackingInitOptions): PreprocessorGroup {
+  const mergedOptions = { ...defaultComponentTrackingOptions, ...options };
+
+  const visitedFiles = new Set<string>();
+
+  return {
+    // This script hook is called whenever a Svelte component's <script>
+    // content is preprocessed.
+    // `content` contains the script code as a string
+    script: ({ content, filename, attributes }) => {
+      // TODO: Not sure when a filename could be undefined. Using this 'unknown' fallback for the time being
+      const finalFilename = filename || 'unknown';
+
+      if (!shouldInjectFunction(mergedOptions.trackComponents, finalFilename, attributes, visitedFiles)) {
+        return { code: content };
+      }
+
+      const { trackInit, trackUpdates } = mergedOptions;
+      const trackComponentOptions: TrackComponentOptions = {
```

Comment:
I would be very specific with the type here since we call JSON.stringify on it. If for some reason we decide to introduce a function option, or even circular options, this will get weird.

Reply:
But we are already very specific with this, right? `mergedOptions` is typed to accept booleans, as well as `TrackComponentOptions`. Am I missing something?

Reply:
Right. I think I confused things a bit. The only thing I am worried here, is that in the future if we change, for example, `trackUpdates` in `TrackComponentOptions` to accept a function, things might break since we're trying to stringify that function. Anyhow, I guess tests will catch that, and also it is hard to enfore something like this with just types.
You can see this issue as resolved :)

Reply:
Ahh I see, good point! But yeah, let's revisit this if it comes to that